linear_algebra/src/conjugate_gradient.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+
+
+def _is_matrix_spd(A: np.array) -> bool:
+
+    """
+    Returns True if input matrix A is symmetric positive definite.
+    Returns False otherwise.
+
+    For a matrix to be SPD, all eigenvalues must be positive.
+
+    >>> import numpy as np
+    >>> A = np.array([
+    ... [4.12401784, -5.01453636, -0.63865857],
+    ... [-5.01453636, 12.33347422, -3.40493586],
+    ... [-0.63865857, -3.40493586,  5.78591885]])
+    >>> _is_matrix_spd(A)
+    True
+    >>> A = np.array([
+    ... [0.34634879,  1.96165514,  2.18277744],
+    ... [0.74074469, -1.19648894, -1.34223498],
+    ... [-0.7687067 ,  0.06018373, -1.16315631]])
+    >>> _is_matrix_spd(A)
+    False
+    """
+    # Ensure matrix is square.
+    assert np.shape(A)[0] == np.shape(A)[1]
+
+    # Get eigenvalues and eignevectors for a symmetric matrix.
+    eigen_values, _ = np.linalg.eigh(A)
+
+    # Check sign of all eigenvalues.
+    if np.all(eigen_values > 0):
+        return True
+    else:
+        return False
+
+
+def _create_spd_matrix(N: np.int64) -> np.array:
+    """
+    Returns a symmetric positive definite matrix given a dimension.
+
+    Input:
+    N is an integer.
```

Comment:
This line is redundant now that we have type hints so let's instead give `N` a more self-documenting variable name so that this comment is no longer required or make this comment document the what & why, not the datatype.

Reply:
Yes good point. I changed this.